packages/agents/sequencer/src/lib/operations/auctions.ts

```diff
@@ -193,139 +192,187 @@ export const executeAuctions = async (_requestContext: RequestContext) => {
+                );
+                if (!routerLiquidity) {
+                  // Either we haven't cached the liquidity yet, or the value cached has become expired.
+                  routerLiquidity = await subgraph.getAssetBalance(destination, router, asset);
+                  if (!routerLiquidity.eq(constants.Zero)) {
+                    await cache.routers.setLiquidity(router, destination, asset, routerLiquidity);
+                  } else {
+                    // NOTE: Using WARN level here as this is unexpected behavior... routers who are bidding on a transfer should
+                    // have added liquidity for the asset on the corresponding domain.
+                    logger.warn(
+                      "Skipped bid from router; liquidity not found in subgraph",
+                      requestContext,
+                      methodContext,
+                      {
+                        transfer: {
+                          transferId,
+                          asset,
+                          destination,
+                          amount: amount.toString(),
+                        },
+                        assetBalanceId: `${asset.toLowerCase()}-${router.toLowerCase()}`,
+                        routerLiquidity,
+                        router,
+                      },
+                    );
+                    insufficientRouterExist = true;
+                    break;
+                  }
+                }
+
+                routerLiquidityMap.set(router, routerLiquidity);
+
+                if (routerLiquidity.lt(assignedAmount)) {
+                  logger.info("Skipped bid from router: insufficient liquidity", requestContext, methodContext, {
+                    transfer: {
+                      transferId,
+                      asset,
+                      destination,
+                      totalAmount: amount.toString(),
+                      assignedAmount,
+                    },
+                    router,
+                    liquidity: routerLiquidity.toString(),
+                  });
+                  insufficientRouterExist = true;
+                  break;
+                }
               }
-            }
 
-            if (routerLiquidity.lt(amount)) {
-              logger.info("Skipped bid from router: insufficient liquidity", requestContext, methodContext, {
-                transfer: {
+              // Skip this combination if there is a router which doesn't have enough liquidity
+              if (insufficientRouterExist) continue;
+
+              try {
+                logger.debug("Sending bid to relayer", requestContext, methodContext, {
                   transferId,
-                  asset,
+                  bid: {
+                    // NOTE: Obfuscating signatures here for safety.
+                    routers: randomCombination.map((bid) => bid.router),
+                    fee: randomCombination.map((bid) => bid.fee),
+                    totalFee: randomCombination.reduce(
```

Comment:
Again, can probably remove this `fee` stuff here. But ultimately, it's supposed to be a percentage (like `"5"` basis points aka 0.05%), *not* a specific value. So definitely should remove `totalFee` as makes no sense to add those percentages together lol.